### DISCUSSION TITLE
Is there a way to see elapsed time in history?

### DISCUSSION THREAD

**Discussion by pgr0ss** [fish-shell/fish-shell#10743] [Q&A]
zsh has a nice feature where I can see the elapsed time of commands as part of the history (with the `-D` flag). For example:

```
% history -D
10002  0:01  sleep 1
10003  0:03  sleep 3
10004  0:05  sleep 5
```

I know fish has `history -t` to see when the command was run, but is there something similar that shows how long each command took? Thanks!

**Comment by nenahp** [comment]
possible with plugin，e.g. https://github.com/atuinsh/atuin

**Reply by pgr0ss** [reply]
Yeah, I've seen `atuin` but that's a pretty drastic change. I was wondering if there was something built in or simpler.

**Comment by faho** [comment]
Fish does not currently store the duration in the history.

I guess we could add it, but tbh I doubt it would be used a lot or is all that helpful.
